code/datums/elements/mirage_border.dm

```diff
@@ -0,0 +1,42 @@
+/**
+ * Creates a mirage effect allowing you to see around the world border, by adding the opposite side to its vis_contents.
+ */
+/datum/element/mirage_border
+
+/datum/element/mirage_border/Attach(datum/target, turf/target_turf, direction, range=world.view)
+	. = ..()
+	if(!isturf(target))
+		return ELEMENT_INCOMPATIBLE
+	if(!target_turf || !istype(target_turf) || !direction)
+		. = ELEMENT_INCOMPATIBLE
+		CRASH("[type] improperly attached with the following args: target=\[[target_turf]\], direction=\[[direction]\], range=\[[range]\]")
+
+	var/atom/movable/mirage_holder/holder = new(target)
+
+	var/x = target_turf.x
+	var/y = target_turf.y
+	var/z = target_turf.z
+	// Handle "15x15", converting it to 15
+	if(istext(range))
+		range = max(getviewsize(range)[1], getviewsize(range)[2])
+	var/turf/southwest = locate(clamp(x - (direction & WEST ? range : 0), 1, world.maxx), clamp(y - (direction & SOUTH ? range : 0), 1, world.maxy), clamp(z, 1, world.maxz))
+	var/turf/northeast = locate(clamp(x + (direction & EAST ? range : 0), 1, world.maxx), clamp(y + (direction & NORTH ? range : 0), 1, world.maxy), clamp(z, 1, world.maxz))
+	//holder.vis_contents += block(southwest, northeast) // This doesnt work because of beta bug memes
```

Comment:
Do you know when this was added? I see it in the old component, and I have no clue if beta is an adjective or if it's referring to something like 515. if the code works perfectly fine without it i'd rather you just remove the line entirely, or at the bare minimum elaborate on why it needs to be commented out in the explanatory comment.

Reply:
the commented out line shouldn't have passed QA the first time around so it would be appreciated if it could be cleaned up now in this _refactor_

Reply:
I suppose, but I'm not really sure what the original author intended

Reply:
Oh it's literally just the loop below but in a single line format. I'll see if it works.